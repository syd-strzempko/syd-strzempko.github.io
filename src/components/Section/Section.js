import styles from './Section.module.css';

const Section = ({left, right}) => {
  return (
    <div className={styles.section}>
      <div className={styles.left}>
        <div className={styles.body}>{left}</div>
      </div>
      <div className={styles.right}>{right}</div>
    </div>
  );
}

export default Section;