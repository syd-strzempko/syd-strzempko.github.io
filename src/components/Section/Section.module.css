.section {
    min-height: 100vh;
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    justify-content: stretch;
    align-items: stretch;
    gap: 20px;
    .left, .right {
        flex: 1 1 0;
        display: flex;
        flex-flow: column nowrap;
        justify-content: center;
        .body {
            padding: 30px 0 30px 30px;
        }
    }
    .right {
        margin: 0 20px 20px 0;
    }
}

@media screen and (max-width: 600px) {
    .section {
        flex-flow: column nowrap;
        .right {
            margin: 10px 0;
        }
        .left {
            order: 2;
            .body {
                padding: 0 20px 10px;
            }
        }
    }
}