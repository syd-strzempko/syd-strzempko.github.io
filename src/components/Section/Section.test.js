import Section from '../Section';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router

describe('Section', () => {
  it('renders correctly', () => {
    let view = renderer.create(
      <MemoryRouter>
        <Section />
      </MemoryRouter>
    );
    expect(view.toJSON()).toMatchSnapshot();
  });
});