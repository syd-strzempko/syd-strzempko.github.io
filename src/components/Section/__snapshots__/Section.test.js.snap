// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<div
  className="section"
>
  <div
    className="left"
  >
    <div
      className="body"
    />
  </div>
  <div
    className="right"
  />
</div>
`;
